python/lsst/ts/m2gui/model.py

```diff
@@ -55,19 +60,170 @@ def __init__(self, log):
         self.local_mode = LocalMode.Standby
         self.is_closed_loop = False
 
-        self.signal_error = SignalError()
+        self.signal_status = SignalStatus()
+
+        self.system_status = self._set_system_status()
+
+        self.fault_manager = FaultManager(self.get_actuator_default_status(False))
+
+    def _set_system_status(self):
+        """Set the default system status.
+
+        Returns
+        -------
+        system_status : `dict`
+            Default system status.
+        """
+
+        system_status = {
+            "isCrioConnected": False,
+            "isTelemetryActive": False,
+            "isInPosition": False,
+            "isInMotion": False,
+            "isTcsActive": False,
+            "isPowerCommunicationOn": False,
+            "isPowerMotorOn": False,
+            "isOpenLoopMaxLimitsEnabled": False,
+            "isAlarmWarningOn": False,
+        }
+
+        return system_status
+
+    def get_actuator_default_status(self, status):
+        """Get the default actuator status.
+
+        Parameters
+        ----------
+        status : `any`
+            Default status.
+
+        Returns
+        -------
+        collection : `dict`
+            Collection of default actuator status.
+        """
+
+        collection = dict()
+        collection = self._set_actuator_default_status_specific_ring(
+            Ring.B, 30, collection, status
+        )
+        collection = self._set_actuator_default_status_specific_ring(
+            Ring.C, 24, collection, status
+        )
+        collection = self._set_actuator_default_status_specific_ring(
+            Ring.D, 18, collection, status
+        )
+        collection = self._set_actuator_default_status_specific_ring(
+            Ring.A, 6, collection, status
+        )
+
+        return collection
+
+    def _set_actuator_default_status_specific_ring(
+        self, ring, number, collection, status
+    ):
+        """Set the default actuator status in specific ring.
+
+        Parameters
+        ----------
+        ring : enum `Ring`
+            Name of ring.
+        number : `int`
+            Number of actuator in the ring.
+        collection : `dict`
+            Collection of actuator status.
+        status : `any`
+            Default status.
+
+        Returns
+        -------
+        collection : `dict`
+            Updated collection.
+        """
+
+        for idx in range(1, number + 1):
+            name = ring.name + str(idx)
+            collection[name] = status
+
+        return collection
+
+    def add_error(self, error):
+        """Add the error.
+
+        Parameters
+        ----------
+        error : `int`
+            Error code.
+        """
+
+        self.fault_manager.add_error(error)
+        self._check_error_and_update_status()
+
+    def _check_error_and_update_status(self):
+        """Check the error is on or not, and update the internal related
+        status."""
```

Comment:
Better: "Check whenever the error is triggered, and update the related internal status" ?

Reply:
Updated in the commit: 0c06a5b. Thanks!